### DISCUSSION TITLE
New collection review: community.sops

### DISCUSSION THREAD

**Discussion by endorama** [ansible-collections/ansible-inclusion#9] [Resolved reviews]
Hello, I'd like to submit `community.sops` collection for review to have it included in Ansible 3.0.0. 

Following along https://github.com/ansible/community/issues/546#issuecomment-743358625, here the requested information:

[Collection on Galaxy](https://galaxy.ansible.com/community/sops) (version 1.0.2)
Collection Source code: https://github.com/ansible-collections/community.sops
Collection issue tracker: https://github.com/ansible-collections/community.sops/issues
Collection maintainer: @endorama

We tracked [in this milestone](https://github.com/ansible-collections/community.sops/milestone/1) all work to be compliant with requirements.

Thank you for your help and your time.

**Comment by tadeboro** [comment]
Hi @endorama!

The community has agreed on a list of criteria that must be met to move forward with the inclusion of a collection in the Ansible package. For this discussion, we will use a shortened checklist to verify and record compliance, but you can find the full details in the [Collection Requirements](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst) documentation.

This is a new and gradual process, and we will edit this comment as we progress along. The items in the checklist can be in one of the following states:

  * Items with a checkmark and no attached comment are OK and need no intervention from your side.
  * Items with a checkmark that contain a comment will not block the inclusion process, but by following the advice given in the comment, the collection's quality can be improved.
  * Items with no checkmark and no extra details are waiting for evaluation.
  * Items with no checkmark that contain at least one comment need your attention. Issues in comments marked as *required* need to be fixed before collection can be included in Ansible. Comments marked as *optional* are just friendly advice and will not block the inclusion.

Feel free to ask any questions and provide feedback. Thanks in advance for your understanding.

**Public availability and communication:**
- [x] published on Ansible Galaxy [Ansible Galaxy](https://galaxy.ansible.com) with version 1.0.0 or later
- [x] has a policy of releasing, versioning and deprecation announced to contributors and users in some way
- [x] have a Code of Conduct (CoC) compatible with the [Ansible Code of Conduct (CoC)](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html)
- [x] have a publicly available issue tracker that does not require a paid level of service to create an account or view issues

**Standards and documentation:**
- [x] adheres to [semantic versioning](https://semver.org/)
- [x] follows licensing rules
- [x] follows the [Ansible documentation standards](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html) and the [style guide](https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide)
- [x] follows [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html)
- [x] has `README.md`
- [x] documentation, examples, and return use FQCNs for `M(..)`, examples, and `seealso` subsections
- [x] modules (or plugins) from ansible-base use `ansible.builtin.` as a FQCN prefix
- [x] documentation and return sections use `version_added:` containing the *collection* version for which an option, module, or plugin was added
- [x] FQCNs is used for `extends_documentation_fragment:`, unless you are referring to doc_fragments from ansible-base

**Collection management:**
- [x] `galaxy.yml` having `tags` field set
- [x] collection dependencies must have a lower bound on the version which is at least 1.0.0 and are all part of the `ansible` package
- [x] `meta/runtime.yml` defines the minimal version of Ansible which the collection works with
- [x] has changelog, preferably with `changelogs/changelog.yaml`

**Tests:**
- [x] passed `ansible-test sanity`
- [x] if `test/sanity/ignore*.txt` exists, it does not contain error codes listed [here](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#ci-testing)
- [x] has CI tests up and running against each of the "major versions" of `ansible-base`/`ansible-core` that the collection supports
- [x] all CI tests run against every pull request
- [x] all CI tests run against a commit that releases the collection; if they don't pass, the collection won't be released
- [x] all CI tests run regularly (nightly, or at least once per week)

**Reply by tadeboro** [reply]
I finished my initial review and found only two blockers for inclusion which should not take much time to resolve.

I did notice that the README file content is a bit outdated in some places (the support matrix is missing at least the latest release, [included content](https://github.com/ansible-collections/community.sops#included-content) section is missing modules). But these minor issues do not block the inclusion process, I just thought that I would point them out since I noticed them during the review process.

All in all, great work, @endorama!

**Reply by felixfontein** [reply]
Nightly CI runs are added by https://github.com/ansible-collections/community.sops/pull/49

**Reply by felixfontein** [reply]
README improvements, including a section on releasing, versioning and deprecation, in https://github.com/ansible-collections/community.sops/pull/50

**Reply by endorama** [reply]
Hi @tadeboro, thank you for the review.

@felixfontein thanks as usual for your fast contributions!

I merged both PR, let me know if anything else need work!

**Reply by tadeboro** [reply]
I checked again two unfulfilled requirements and I am happy to conclude this review with approval. Great work @endorama and @felixfontein!

**Comment by Andersson007** [comment]
@felixfontein i fixed minor doc and flake8 issues https://github.com/ansible-collections/community.sops/pull/52

**Comment by Andersson007** [comment]
**Public availability and communication:**
- [x] published on Ansible Galaxy [Ansible Galaxy](https://galaxy.ansible.com) with version 1.0.0 or later
- [x] has a policy of releasing, versioning and deprecation announced to contributors and users in some way
- [x] have a Code of Conduct (CoC) compatible with the [Ansible Code of Conduct (CoC)](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html)
- [x] have a publicly available issue tracker that does not require a paid level of service to create an account or view issues

**Standards and documentation:**
- [x] adheres to [semantic versioning](https://semver.org/)
- [x] follows licensing rules
- [x] follows the [Ansible documentation standards](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html) and the [style guide](https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide)
- [x] follows [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html)
- [x] has `README.md`
- [x] documentation, examples, and return use FQCNs for `M(..)`, examples, and `seealso` subsections
- [x] modules (or plugins) from ansible-base use `ansible.builtin.` as a FQCN prefix
- [x] documentation and return sections use `version_added:` containing the *collection* version for which an option, module or plugin was added
- [x] FQCNs is used for `extends_documentation_fragment:`, unless you are referring to doc_fragments from ansible-base

**Collection management:**
- [x] `galaxy.yml` having `tags` field set
- [x] collection dependencies must have a lower bound on the version which is at least 1.0.0, and are all part of the `ansible` package
- [x] `meta/runtime.yml` defines the minimal version of Ansible which the collection works with
- [x] has changelog, preferably with `changelogs/changelog.yaml`

**Tests:**
- [x] passed `ansible-test sanity`
- [x] if `test/sanity/ignore*.txt` exists, it does not contain error codes listed [here](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#ci-testing)
- [x] has CI tests up and running against each of the "major versions" of `ansible-base`/`ansible-core` that the collection supports
- [x] all CI tests run against every pull request
- [x] all CI tests run against a commit that releases the collection; if they don't pass, the collection won't be released
- [x] all CI tests run regularly (nightly, or at least once per week)

**Comment by felixfontein** [comment]
@tadeboro created a PR for inclusion in Ansible 3.0.0: https://github.com/ansible-community/ansible-build-data/pull/52

**Reply by dmsimard** [reply]
Pull request has been merged, thank you @tadeboro and @Andersson007 for reviewing !
